server/app/views/components/TextFormatter.java

```diff
@@ -1,190 +1,121 @@
-      if (line.startsWith(TextFormatter.ACCORDION_HEADER)) { // We're calling this an accordion.
-        String accordionHeader = line.substring(3);
-        int next = i + 1;
-        ArrayList<String> content = new ArrayList<>();
-        while (next < lines.length && lines[next].startsWith(TextFormatter.ACCORDION_CONTENT)) {
-          content.add(lines[next].substring(1));
-          next++;
-        }
-        i = next - 1;
-        String accordionContent = content.stream().collect(Collectors.joining("\n"));
-        builder.add(buildAccordion(accordionHeader, accordionContent));
-      } else if (line.startsWith(TextFormatter.BULLETED_ITEM)) { // unordered list item.
-        ArrayList<String> items = new ArrayList<>();
-        items.add(line.substring(1).trim());
-        int next = i + 1;
-        while (next < lines.length && lines[next].startsWith(TextFormatter.BULLETED_ITEM)) {
-          items.add(lines[next].substring(1).trim());
-          next++;
-        }
-        i = next - 1;
-        builder.add(buildList(items));
-      } else if (line.length() > 0) {
-        ImmutableList<DomContent> lineContent =
-            TextFormatter.createLinksAndEscapeText(
-                line, UrlOpenAction.NewTab, /*addRequiredIndicator= */ false);
-        builder.add(div().with(lineContent));
-      } else if (preserveEmptyLines) {
-        builder.add(div().withClasses("h-6"));
+      if (line.isBlank()) {
+        // Add an empty space character so the blank line is preserved
+        lines[i] = "&nbsp;\n";
       }
     }
-    return builder.build();
+    return String.join("\n", lines);
   }
 
-  private static DivTag buildAccordion(String title, String accordionContent) {
-    Accordion accordion = new Accordion().setTitle(title);
-    ImmutableList<DomContent> contentTags = TextFormatter.formatText(accordionContent, true);
-    contentTags.forEach(accordion::addContent);
-    return accordion.getContainer();
+  private static String addIconToLinks(String markdownText) {
+    String closingATag = "</a>";
+    String svgIconString =
+        Icons.svg(Icons.OPEN_IN_NEW)
+            .withClasses("shrink-0", "h-5", "w-auto", "inline", "ml-1", "align-text-top")
+            .toString();
+    return markdownText.replaceAll(closingATag, svgIconString + closingATag);
   }
 
-  private static UlTag buildList(List<String> items) {
-    UlTag listTag = ul().withClasses("list-disc", "mx-8");
-    items.forEach(item -> listTag.with(li().withText(item)));
-    return listTag;
+  private static String addRequiredIndicatorInsidePTag(String markdownText) {
+    int indexOfPTag = markdownText.lastIndexOf("</p>");
+    String stringWithRequiredIndicator = ViewUtils.requiredQuestionIndicator().toString() + "</p>";
+    return markdownText.substring(0, indexOfPTag) + stringWithRequiredIndicator;
+  }
+
+  private static String sanitizeHtml(String markdownText) {
+    PolicyFactory customPolicy =
+        new HtmlPolicyBuilder()
```

Comment:
This will strip any elements and attributes not on the "allowed list" before returning the html. This disallows script tags for example.